### ISSUE TITLE
[css-fonts] Proposal for a new generic font family "Rounded"

### ISSUE THREAD

**Issue by c933103** [w3c/csswg-drafts#4605]
I have noted that in https://www.w3.org/TR/css-fonts-4/#generic-font-families there are now proposal for a generic font fanily known as `ui-rounded`. However, it seems a bit strange to have a rounded font family option to resemble rounded systen UI font yet without an independent rounded font option to use other rounded fonts that do not need to be same as the rounded font in system UI. It would be nice if there can be a separate `rounded` family of font so that there can be more flexibility for users to choose their desired rounded font.

**Comment by jfkthame** [comment]
I'm unconvinced that adding more CSS "generic" font-family names is a good path forward. The set of possible "generic" names for certain classes of design is pretty open-ended (not to mention highly dependent on writing system), and requests will likely be driven as much by the winds of design fashion as by any principled model or taxonomy of fonts.

(As for the justification that this parallels `ui-rounded`: ISTM even the `ui-*` names are a bit problematic, as they may not actually achieve what authors want. Suppose I design a page to match current macOS UI conventions by using `ui-serif`, `ui-rounded`, etc on appropriate elements. But then a future revision of the OS decides to change its typographic conventions so that the elements that in Catalina were using a rounded font now appear in a slab serif instead: my page suddenly becomes a misfit. System UI conventions have changed in the past, and it seems highly likely they will change in the future. For an author who wants to match the system UI design, it seems like the old system font keywords (`caption`, `icon`, etc) would be a more reliable approach than directly calling for `ui-rounded`, `ui-serif`, etc.)

**Comment by litherum** [comment]
I proposed a set of criteria for adding new generic font families in https://github.com/w3c/csswg-drafts/issues/4910. Perhaps we should finish determining these criteria before deciding on this issue.

**Comment by litherum** [comment]
> OS decides to change its typographic conventions so that the elements that in Catalina were using a rounded font now appear in a slab serif

I don't think this problem is specific to the `ui-` generic families. An OS can replace Times with a sans-serif if they feel like it, and the probability an OS would actually do this is about the same probability that an OS would replace `ui-rounded` with a slab serif.

**Comment by kidayasuo** [comment]
“rounded” will be a valuable addition for Japan. I do not have a hard number but it is most likely be the most popular typeface category after serif and sans-serif (c.f. links to font category list by two largest Japanese type foundries below). It is an important typeface in DTP and e-books. Both macOS/iOS and Windows (through Office) have the typeface.

It exists in wide range of languages such as Latin, Chinese (Yuan Ti 圆体), Arabic, Thai and I believe in many other languages. Most importantly they are in a consistent style across languages, i.e. sans-serif with rounded terminals. It is one of the few modern, or abstract, typefaces like sans-serif that do not carry much cultural dependency.

**Comment by AmeliaBR** [comment]
Some things to consider here:

- Would any user agent expose a different font for `rounded` vs `ui-rounded`?
- Would they allow one to be user-customized and not the other?
- Are there user agents / platforms that have rounded fonts but not something that fits the definition of a `ui-rounded` font (because it's not used in the user interface and doesn't match the other UI fonts)?

When the `ui-*` fonts were introduced, the argument was that there were legacy expectations preventing `serif` and `sans-serfi` from being updated to anything other than Times and Arial/Helvetica. With that perspective, the `ui-*` generics are the “modern” generics, and the other keywords are legacy fallbacks. There is no legacy argument for a `rounded`. But the `ui` prefix implies extra constraints and expectations that might not be appropriate if an author just wants a rounded style for big headings or buttons, without trying to mimic the UI at all.

Also, re:

>> OS decides to change its typographic conventions
>
> the same probability that an OS would replace ui-rounded with a slab serif. (That is to say, ~0 probability.)

I think you've misinterpreted the concern, @litherum.  The complaint against the `ui` fonts is that _if_ their purpose was to mimic the native UI, you could nonetheless end up with a dated design if a future design trend means that rounded fonts are used for headings (instead of sans-serif) and serif fonts for buttons and so on.

But that's separate from the question of whether there is any benefit to a non-ui generic.

**Comment by fantasai** [comment]
Summary of the CSSWG discussion was that there was broad consensus to add this, but some concerns about the privacy impact due to fingerprinting. So if we can be convinced that this concern is resolvable, we will (presumably) add a new generic for rounded sans-serif fonts.

Wrt naming, @LeaVerou suggested `rounded-sans`, which has the benefit over `rounded` of being less likely to trigger some kind of naming clash.

**Comment by jfkthame** [comment]
Wasn't there also a higher-level question re syntax for new generics: whether to simply add keywords (which always have the potential to clash with actual font-family names -- an increasing problem if we're going to keep extending the set), or to introduce something distinct such as a `generic(...)` function?

**Comment by r12a** [comment]
By my reading, this was only an agreement in principle that a 'rounded' generic was useful, and that further discussion was needed on what generics are for and what syntax to use, before finalising the discussion.

**Comment by LeaVerou** [comment]
> Wrt naming, @LeaVerou suggested `rounded-sans`, which has the benefit over `rounded` of being less likely to trigger some kind of naming clash.

I did suggest `rounded-sans` but not exactly in that context. What I was suggesting was to have generics that are subcategories of other generics, e.g. `rounded-sans`, `slab-serif` etc. Possibly even a function with keyword traits e.g. `generic(sans-serif rounded)` or `generic(old-style serif)` etc, which could fallback to fewer keywords instead of just whatever's next on the stack.
